components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java

```diff
@@ -59,9 +61,39 @@ public ConfigReader generateConfigReader(String namespace, String name) {
             }
         }
         if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Couldn't find a matching configuration for name: "+
+            LOGGER.debug("Couldn't find a matching configuration for name: " +
                     name + "and namespace: " + namespace + "!");
         }
         return new FileConfigReader(new HashMap<>());
     }
+
+    @Override
+    public Map<String, String> extractStoreConfigs(String name) {
+        if (configProvider != null) {
+            try {
+                RootConfiguration rootConfiguration = configProvider.getConfigurationObject(RootConfiguration.class);
+                if (null != rootConfiguration && null != rootConfiguration.stores) {
+                    for (Store store : rootConfiguration.stores) {
+                        StoreChildConfiguration childConfiguration = store.getStore();
+                        if (null != childConfiguration && null != childConfiguration.getName()
+                                && childConfiguration.getName().equals(name)) {
+                            Map<String, String> storeConfigs = new HashMap<>();
+                            storeConfigs.put(SiddhiConstants.ANNOTATION_ELEMENT_TYPE, childConfiguration.getType());
+                            if (childConfiguration.getProperties() != null) {
+                                storeConfigs.putAll(childConfiguration.getProperties());
+                            }
+                            return storeConfigs;
+                        }
+                    }
+                }
+            } catch (CarbonConfigurationException e) {
+                LOGGER.error(e.getMessage(), e);
+            }
+        }
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Couldn't find a matching configuration for store, name: " +
+                    name + "!");
+        }
+        return new HashMap<>();
```

Comment:
Does the upstream code handle empty HashMap scenario?

Reply:
Yes, it does. (https://github.com/wso2/siddhi/blob/master/modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/helper/DefinitionParserHelper.java#L833)